Sources/TuistKit/ProjectGenerator/ProjectGenerator.swift

```diff
@@ -220,7 +220,17 @@ class ProjectGenerator: ProjectGenerating {
         // Apply graph mappers
         let (updatedGraph, graphMapperSideEffects) = try graphMapperProvider.mapper(config: config).map(graph: graph)
 
-        return (workspace, updatedGraph, modelMapperSideEffects + graphMapperSideEffects)
+        /// Used to disambiguate which workspaces are generated by tuist and which are not
+        let tuistGeneratedFileDescriptor = FileDescriptor(
+            path: workspace.path.appending(components: "\(workspace.name).xcworkspace", Constants.tuistGeneratedFileName)
+        )
+
+        /// Additional side effects
+        let additionalSideEffects: [SideEffectDescriptor] = [
+            .file(tuistGeneratedFileDescriptor),
+        ]
+
+        return (workspace, updatedGraph, modelMapperSideEffects + graphMapperSideEffects + additionalSideEffects)
```

Comment:
Note: We're in the `loadWorkspace` - this will only add a `.tuist-generated` to projects with a `Workspace.swift`, projects with a single `Project.swift` won't get it, is that desired?

There are three primary methods in here `generateProject`, `generateProjectWorkspace` and `generateWorkspace`

- `generateProject` > `loadProject` > writes a single project descriptor (`.xcodeproj`)
- `generateProjectWorkspace` > `loadProject` > writes workspace descriptor (`.xcworkspace` + `.xcodeproj`s)
- `generateWorkspace` > `loadWorkspace` > writes workspace descriptor (`.xcworkspace` + `.xcodeproj`s)

I believe the intention is to add the side effect in both `generateProjectWorkspace` and `generateWorkspace`. To get the side effect to both some options:

- add `loadProjectWorkspace` (which calls `loadPorject` under the hood + adds additional side effects)
- add a `write(workspaceDescriptor:)` which creates a new descriptor that includes the side effect ahead of calling `writer.write(..)` - this ensures all workspace writes within here include that side effect.

Reply:
Currently, `BuildGraphInspector` only leverages `.xcworkspace`, so I'd maybe be in favor of not adding it to `.xcodeproj` for now until it's necessary to do so - what do you think?

Reply:
Sure makes sense to add only what’s required 👍

Reply:
If understood @kwridan's point properly, we should add it to all the methods that generate workspaces: `generateProjectWorkspace`, & `generateWorkspace`. Although it's not necessary yet, something tells me that is a good idea to add it to the generated Xcode projects too. What do you think?

Reply:
I agree, I misunderstood it; I already talked it over with @kwridan and will add it, forgot to add the result of that discussion here 🙂